File: testdata/fixtures/php/vulnerable/eval_injection.php
<?php
// Code injection via eval() with user-controlled input.
// Vulnerable: eval executes arbitrary PHP code.

// GTSS-INJ-003: eval() with variable from user input
if (isset($_GET['code'])) {
    $code = $_GET['code'];
    eval($code);
}

// Calculator-style eval injection
if (isset($_POST['expression'])) {
    $expr = $_POST['expression'];
    // Attacker can inject: 1; system('id');
    eval('$result = ' . $expr . ';');
    echo "Result: " . $result;
}

// Dynamic class method invocation via eval
$method = $_GET['method'];
$args = $_GET['args'];
eval("\$obj->$method($args);");

// assert() can also execute code in older PHP versions
if (isset($_GET['check'])) {
    $check = $_GET['check'];
    assert($check);
}

// Variable variables - indirect code execution
$var_name = $_GET['var'];
$$var_name = $_GET['value'];

File: testdata/fixtures/python/vulnerable/eval_injection.py
# Vulnerable: Code injection via eval() with user input
# Expected: GTSS-INJ-003 (Code Injection)

from flask import Flask, request, jsonify

app = Flask(__name__)


@app.route("/calculate")
def calculate():
    expression = request.args.get("expr")
    result = eval(expression)
    return jsonify({"result": result})


@app.route("/filter", methods=["POST"])
def dynamic_filter():
    filter_expr = request.form.get("filter")
    data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
    filtered = eval(f"[x for x in data if {filter_expr}]")
    return jsonify({"filtered": filtered})


@app.route("/transform", methods=["POST"])
def transform():
    code = request.form.get("code")
    data = request.json.get("data", [])
    exec(code)
    return jsonify({"status": "executed"})


@app.route("/config", methods=["POST"])
def set_config():
    key = request.form.get("key")
    value = request.form.get("value")
    eval(f"config.{key} = {value}")
    return "OK"
